plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/DefaultTextKafkaProducerFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kafka;
+
+import io.trino.spi.connector.ConnectorSession;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultTextKafkaProducerFactory
+        implements KafkaProducerFactory
+{
+    private final Properties externalProperties;
```

Comment:
It is better to use `Map<String,String>` as this could be effectively immutable.